devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy

```diff
@@ -14,181 +13,321 @@
+            state "default", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        main (["button"])
+        details(["button", "battery", "temperature", "refresh"])
+    }
 }
 
 def parse(String description) {
-    def descMap = zigbee.parseDescriptionAsMap(description)
-    logIt descMap
-    state.lastCheckin = now()
-    logIt "lastCheckin = ${state.lastCheckin}"
-    handlePresenceEvent(true)
-    
-	def results = []
-    if (description?.startsWith('catchall:'))
-		results = parseCatchAllMessage(descMap)
-	else if (description?.startsWith('read attr -'))
-		results = parseReportAttributeMessage(descMap)
-    else logIt(descMap, "trace")
-        
-	return results;
-}
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        sendEvent(event)
+    }
+    else {
+        if ((description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))) {
+            def descMap = zigbee.parseDescriptionAsMap(description)
+            if (descMap.clusterInt == 0x0402 && descMap.attrInt == 0) {
+                event = parseTemperature(descMap.value)
+            }
+            else if (descMap.clusterInt == 0x0001 && descMap.attrInt == 0x0020) {
+                event = getBatteryResult(zigbee.convertHexToInt(descMap.value))
+            }
+            else if (descMap.clusterInt == 0x0006 || descMap.clusterInt == 0x0008) {
+                event = parseNonIasButtonMessage(descMap)
+            }
+        }
+        else if (description?.startsWith('temperature: ')) {
+            def value = zigbee.parseHATemperatureValue(description, "temperature: ", getTemperatureScale())
+            event = getTemperatureResult(value)
+        } else if (description?.startsWith('zone status')) {
+            event = parseIasButtonMessage(description)
+        }
 
-def updated() {
-	startTimer()
-    configure()
+        log.debug "Parse returned $event"
+        def result = event ? createEvent(event) : null
+
+        if (description?.startsWith('enroll request')) {
+            List cmds = enrollResponse()
+            result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+        }
+        return result
+    }
 }
 
-def configure(){
-	logIt "Configuring Smart Fob..."
-	[
-	"zdo bind 0x${device.deviceNetworkId} 1 1 6 {${device.zigbeeId}} {}", "delay 200",
-    "zdo bind 0x${device.deviceNetworkId} 2 1 6 {${device.zigbeeId}} {}", "delay 200",
-    "zdo bind 0x${device.deviceNetworkId} 3 1 6 {${device.zigbeeId}} {}", "delay 200",
-    "zdo bind 0x${device.deviceNetworkId} 4 1 6 {${device.zigbeeId}} {}", "delay 200",
-    "zdo bind 0x${device.deviceNetworkId} 1 1 1 {${device.zigbeeId}} {}", "delay 200"
-    ] +
-    zigbee.configureReporting(0x0001,0x0020,0x20,20,20,0x01)
+private Map parseIasButtonMessage(String description) {
+    List parsedMsg = description.split(' ')
+    String msgCode = parsedMsg[2]
```

Comment:
Recommend converting to an integer and then looking at the value of bit 1 to determine button press. That way it will still work if other bits in the zone status are set.